tests/vmi_bridge_test.go

```diff
@@ -0,0 +1,390 @@
+	"kubevirt.io/kubevirt/pkg/kubecli"
+
+	"kubevirt.io/kubevirt/pkg/log"
+	"kubevirt.io/kubevirt/tests"
+)
+
+var _ = Describe("Bridge", func() {
+
+	bridgeIP := map[string]string{"red": "172.16.98.1", "blue": "172.16.99.1"}
+
+	flag.Parse()
+
+	virtClient, err := kubecli.GetKubevirtClient()
+	tests.PanicOnError(err)
+	tests.BeforeAll(func() {
+		waitForPodToFinish := func(pod *k8sv1.Pod) k8sv1.PodPhase {
+			Eventually(func() k8sv1.PodPhase {
+				j, err := virtClient.Core().Pods(tests.NamespaceTestDefault).Get(pod.ObjectMeta.Name, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return j.Status.Phase
+			}, 60*time.Second, 1*time.Second).Should(Or(Equal(k8sv1.PodSucceeded), Equal(k8sv1.PodFailed)))
+			j, err := virtClient.Core().Pods(tests.NamespaceTestDefault).Get(pod.ObjectMeta.Name, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			return j.Status.Phase
+		}
+
+		addBridgeToHost := func(name string) {
+			// create bridge on the node
+			parameters := []string{"link", "add", name, "type", "bridge"}
+			job := tests.RenderIPRouteJob(fmt.Sprintf("ip-add-%s", name), parameters)
+
+			// make sure that both jobs are happening on the same node
+			listOptions := metav1.ListOptions{}
+			nodeList, err := virtClient.CoreV1().Nodes().List(listOptions)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(nodeList.Items).NotTo(HaveLen(0))
+			node := nodeList.Items[0]
+			nodeSelector := map[string]string{"kubernetes.io/hostname": node.Name}
+
+			job.Spec.NodeSelector = nodeSelector
+			job, err = virtClient.CoreV1().Pods(tests.NamespaceTestDefault).Create(job)
+			Expect(err).ToNot(HaveOccurred())
+			waitForPodToFinish(job)
+			// dont check results, as this may fail because bridge is already there
+			// if there was any issue with creating the bridges the following "set" command would indicate the failure
+
+			// set the bridge to "up" mode
+			parameters = []string{"link", "set", "dev", name, "up"}
+			job = tests.RenderIPRouteJob(fmt.Sprintf("ip-set-%s", name), parameters)
+			job.Spec.NodeSelector = nodeSelector
+			job, err = virtClient.CoreV1().Pods(tests.NamespaceTestDefault).Create(job)
+			Expect(err).ToNot(HaveOccurred())
+			phase := waitForPodToFinish(job)
+			Expect(phase).To(Equal(k8sv1.PodSucceeded))
+
+			// set IP on bridge
+			parameters = []string{"addr", "add", bridgeIP[name] + "/24", "dev", name}
+			job = tests.RenderIPRouteJob(fmt.Sprintf("ip-addr-%s", name), parameters)
+			job.Spec.NodeSelector = nodeSelector
+			job, err = virtClient.CoreV1().Pods(tests.NamespaceTestDefault).Create(job)
+			Expect(err).ToNot(HaveOccurred())
+			phase = waitForPodToFinish(job)
+			// don't check address setting result
+		}
+
+		// add red and blue bridges to host
+		addBridgeToHost("red")
+		addBridgeToHost("blue")
+	})
+
+	Context("Exposing a network to the VM via bridge device plugin", func() {
+		var vmi *v1.VirtualMachineInstance
+		const networkName = "red"
+		const macAddress = "de:ad:00:00:be:af"
```

Comment:
This is used to assertion, but never set as an parameter, maybe one of the problems is here? I guess this patch must be rebased on the #1430

Reply:
It has additional problems, but all except MAC address must be fixed under https://github.com/kubevirt/kubevirt/pull/1437